packages/ui-icons-lucide/src/wrapLucideIcon/index.tsx

```diff
@@ -0,0 +1,230 @@
+      bidirectional = true,
+      inline = true,
+      title,
+      description,
+      elementRef,
+      themeOverride,
+      absoluteStrokeWidth,
+      className,
+      style,
+      ...rest
+    } = props
+
+    const theme = useTheme() as Theme
+    const iconTheme = theme?.newTheme?.components?.Icon
+
+    const handleElementRef = (el: SVGSVGElement | null) => {
+      if (typeof elementRef === 'function') {
+        elementRef(el)
+      }
+    }
+
+    // Convert semantic size to pixels for Lucide
+    const isSemanticSize = typeof instUISize === 'string'
+    const semanticSize = isSemanticSize ? instUISize : undefined
+    let numericSize: number | undefined
+    if (isSemanticSize && iconTheme) {
+      const sizeMap = {
+        xs: iconTheme.sizeXs,
+        sm: iconTheme.sizeSm,
+        md: iconTheme.sizeMd,
+        lg: iconTheme.sizeLg,
+        xl: iconTheme.sizeXl,
+        '2xl': iconTheme.size2xl
+      }
+      numericSize = px(sizeMap[instUISize as keyof typeof sizeMap])
+    } else if (!isSemanticSize) {
+      numericSize = instUISize
+    }
+
+    // Convert semantic strokeWidth to pixels for Lucide
+    const isSemanticStrokeWidth =
+      typeof instUIStrokeWidth === 'string' &&
+      SEMANTIC_STROKE_WIDTHS.includes(
+        instUIStrokeWidth as (typeof SEMANTIC_STROKE_WIDTHS)[number]
+      )
+    let numericStrokeWidth: number | string | undefined
+    if (isSemanticStrokeWidth && iconTheme) {
+      const strokeWidthMap = {
+        xs: iconTheme.strokeWidthXs,
+        sm: iconTheme.strokeWidthSm,
+        md: iconTheme.strokeWidthMd,
+        lg: iconTheme.strokeWidthLg,
+        xl: iconTheme.strokeWidthXl,
+        '2xl': iconTheme.strokeWidth2xl
+      }
+      numericStrokeWidth = px(
+        strokeWidthMap[instUIStrokeWidth as keyof typeof strokeWidthMap]
+      )
+    } else if (!isSemanticStrokeWidth) {
+      numericStrokeWidth = instUIStrokeWidth
+    }
+
+    // Determine if color is semantic (theme token) or custom CSS
+    const isSemanticColor = instUIColor
+      ? SEMANTIC_COLORS.includes(
+          instUIColor as (typeof SEMANTIC_COLORS)[number]
+        )
+      : false
+    const colorValue = isSemanticColor ? instUIColor : undefined
+    const customColor = isSemanticColor ? undefined : instUIColor
+
+    const styles = useStyle({
+      componentId: 'Icon' as const,
+      generateStyle,
+      params: {
+        size: semanticSize,
+        color: colorValue,
+        rotate,
+        bidirectional,
+        inline,
+        themeOverride
+      },
+      displayName: `LucideIcon(${Icon.displayName || Icon.name})`
+    })
+
+    const accessibilityProps: Record<string, string | boolean> = {}
+    if (title) {
+      accessibilityProps['aria-label'] = description
+        ? `${title}: ${description}`
+        : title
+      accessibilityProps['role'] = 'img'
+    } else {
+      accessibilityProps['aria-hidden'] = 'true'
+      accessibilityProps['role'] = 'presentation'
+    }
```

Comment:
Based on this code, `description` is not a needed field. I'd remove it (and later write a codemod, that converts it to the `title` prop